src/Stream/DynamicMemoryWriter.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "BiDirectionalSeekableWriter.h"
+#include "MemoryReader.h"
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+namespace Stream
+{
+	class DynamicMemoryWriter : public BidirectionalSeekableWriter
+	{
+	public:
+		DynamicMemoryWriter();
+		DynamicMemoryWriter(std::size_t preallocateSize);
+
+		uint64_t Length() override;
+		uint64_t Position() override;
+
+		// Note: SeekForward will set a new stream size and 0 fill the buffer gap
```

Comment:
What do you think about pulling 'Note:' out of these comments? I think it is superfluous in context.

Reply:
You're probably right. I do overuse "Note:" a bit.